File: ruby/all-your-base/all_your_base.rb
class BaseConverter
  def self.convert(input_base, digits, output_base)
    unless input_base > 1 && output_base > 1 && digits.all? { |d| d >= 0 && d < input_base }
      raise ArgumentError
    end
    number = 0
    digits.each { |digit| number = input_base * number + digit }
    digits_out = []
    while number > 0
      digits_out = digits_out.push(number % output_base)
      number = number / output_base
    end
    digits_out.length > 0 ? digits_out.reverse : [0]
  end
end

File: csharp/all-your-base/AllYourBase.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class AllYourBase
{
    public static int[] Rebase(int inputBase, int[] inputDigits, int outputBase)
    {
        if (inputBase < 2) throw new ArgumentException("input base must be >= 2");
        if (outputBase < 2) throw new ArgumentException("output base must be >= 2");
        if (inputDigits.Any(d => d < 0 || d >= inputBase)) 
            throw new ArgumentException("all digits must satisfy 0 <= d < input base");

        var number = inputDigits.Aggregate(0, (current, digit) => current * inputBase + digit);
        List<int> result = new();
        while (number >= outputBase)
        {
            var digit = number % outputBase;
            result.Add(digit);
            number = (number - digit) / outputBase;
        }
        result.Add(number);
        result.Reverse();
        return result.ToArray();
    }
}
